python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -338,10 +337,12 @@ def set_object_dtypes_from_pa_schema(df, schema):
     # "object" dtypes to agree with a specific
     # pyarrow schema.
     if schema:
-        for name in df.columns:
-            if name in schema.names and df[name].dtype == "O":
-                df[name] = df[name].astype(
-                    cudf_dtype_from_pa_type(schema.field(name).type)
+        for col_name, col in df._data.items():
+            if col_name in schema.names and isinstance(
+                col, cudf.core.column.StringColumn
+            ):
```

Comment:
Cool. This seems better than the changes in #9159 - Perhaps we can copy in the test from that PR?

Reply:
Thanks @rjzamora ! Added the test.